### PR TITLE
setup applicant info page and test

### PR DESCRIPTION
**Pull request by rrmatthews**
These changes set up the applicant information page for the complaint tool as described by this [ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11413)

## TODO:
- [x]  write unit tests
  - [x] submit without required information fails
  - [x] submit with required information succeeds
  - [x]  myself (select isVeteran to reveal service history)
  -  [x] someone else option
  - [x] anonymous
- [x] implement conditional logic for each case (of on behalf of options)
- [x] use common definitions where applicable (name, dob)

## Notes: 
-The [figma design](https://www.figma.com/file/FHoTbqOpW2ymjj3xo1V1gU0C/GI-Bill-Complaint-Tool-Userflow) has changed to match the data required by the backend. these screenshots are in line with those changes
-We are setting up a ticket to handle the styling errors of the blue description boxes when you select who you are filing onBehalfOf
-the tool name has been changed and there will be changes to  incorporate "feedback" as opposed to "complaint". Currently waiting for those updates and suggestions.

### Myself
localhost_3001_education_complaint-tool_form_benefits-information

### Someone else
localhost_3001_education_complaint-tool_form_benefits-information 1

### Anonymous
localhost_3001_education_complaint-tool_form_applicant-information

### CODE REVIEW

**Comment by lalexanderson-dsva** [comment]
Question: The title is "opt out of information sharing" which I thought was Colmery Act work Emily was doing, but the first page is the complaint tool. Are we conflating two different tools, or am I very very confused?

**Comment by rrmatthews** [comment]
@lalexanderson-dsva There is another [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/8018) dealing with the code / content you're referencing. I'm certain I made a note of that in here but it seems like those didn't save, so I'll add them again to the initial comment at the top

**Comment by lalexanderson-dsva** [comment]
Ok, so in the screen shots above, we have the title for the Colmery Act Opt-Out Tool but the body of the GI Bill Complaint Tool.

**Review by peggygannon — CHANGES_REQUESTED** [review]
@rrmatthews These screenshots don't reflect some of my original edits that I put into the Figma prototype. 
Also the header is not correct.

**Comment by cehsu** [comment]
@lalexanderson-dsva could you please provide context on why service affiliation was made into a boolean? it looks like the BE is expecting a string with one of the values provided in the dropdown.

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 1]
```diff
@@ -24,18 +57,174 @@ const formConfig = {
...
+                'enum': [
+                  'Myself',
+                  'Someone else',
+                  'I want to submit my complaint anonymously'
+                ]
+              },
+              fullName: {
+                type: 'object',
+                properties: {
+                  // prefix: {
```
Comment (cehsu) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Let's go ahead and enable this, and include prefix in the uiSchema definition above as:
  ```
prefix: {
    'ui:title': 'Prefix',
    'ui:options': {
      widgetClassNames: 'form-select-medium'
    }
  }
```

**Review by cehsu — APPROVED** [review]
LGTM, here is a [ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11619) to capture the style bug fix work. There is also a linter error to resolve: 
```
/application/src/applications/edu-benefits/tests/complaint-tool/config/applicantInfo.unit.spec.js

  6:28  error  'selectCheckbox' is defined but never used  no-unused-vars



✖ 1 problem (1 error, 0 warnings)

```

**Comment by lalexanderson-dsva** [comment]
@cehsu - I don't have that context. @miamay345 may be better suited to answer that question for you.

**Comment by rrmatthews** [comment]
@cehsu I believe this was because the information only matters if the user is a veteran. Essentially it is for user lookup and verification purposes inside VA databases to check that the user is actually a veteran. That's about it, to my understanding. Knowing if the user is a dependent or a spouse, isn't as important as knowing if they're a veteran or not

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 2]
```diff
@@ -24,18 +57,174 @@ const formConfig = {
...
+        applicantInformation: {
+          path: 'applicant-information',
+          title: 'Applicant Information',
           uiSchema: {
+            onBehalfOf: {
+              'ui:widget': 'radio',
+              'ui:title': 'I’m filing on behalf of...',
+              'ui:options': {
+                nestedContent: {
+                  Myself: () => <div className="usa-alert-info no-background-image"><i>(We’ll only share your name with the school.)</i></div>,
```
Comment (rianfowler) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: object keys should be camel case

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 3]
```diff
@@ -24,18 +57,174 @@ const formConfig = {
...
+                  'Someone else': () => <div className="usa-alert-info no-background-image"><i>(We’ll only share your name with the school.)</i></div>,
+                  'I want to submit my complaint anonymously': () => <div className="usa-alert-info no-background-image"><i>(Your personal information won’t be shared with anyone outside of VA.)</i></div>
```
Comment (rianfowler) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Prefer these keys to be camel case properties but if we want to use strings they should be set as constants.

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 1 above)
Comment (rrmatthews) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Okay, so are we implementing this in design as well? Just trying to make some continuity checks..

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 4]
```diff
@@ -24,18 +57,174 @@ const formConfig = {
   defaultDefinitions: {
```
Comment (rianfowler) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Do we need this if it's an empty object?

**Review by rianfowler — APPROVED** [review]
Just a couple of nits but lgtm

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 2 above)
Comment (rrmatthews) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: done!

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 3 above)
Comment (rrmatthews) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: done!

**Comment by cehsu** [comment]
@rrmatthews @miamay345 I asked stakeholders about this yesterday and the indicated that the service affiliation /isVeteran question is only to provide context, and isn't used to determine anything. Should we follow up again?

**Comment by cehsu** [comment]
@rrmatthews could you please ping @peggygannon in a comment on this PR that includes the updated screenshots, once her comments have been incorporated?

**Comment by peggygannon** [comment]
@cehsu @rrmatthews hi, Feedback is misspelled in the header. 

Also, this screen doesn't match the most recent prototype. Can you please pull the most recent copy from the prototype?

**Comment by cehsu** [comment]
@miamay345 stakeholders have indicated that they would like to know the specific service affiliation of the applicant, from one of the following options: 'Service Member', 'Spouse or Family Member', 'Veteran', or 'Other', rather than simply yes/no "is a Veteran or Service Member". Should that list be presented as radio buttons or as a dropdown?

**Comment by cehsu** [comment]
Thanks @peggygannon, we've updated this based on the current design, @rrmatthews could you please provide an updated screenshot for @peggygannon to review?

**Comment by rrmatthews** [comment]
@peggygannon @cehsu screenshot is updated

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 5]
```diff
@@ -27,18 +65,183 @@ const formConfig = {
...
+                expandUnderCondition: isNotAnonymous
+              }
+            },
+            serviceAffiliation: { // could wrap service info in an object
+              'ui:title': 'Service affiliation',
+              'ui:options': {
+                expandUnder: 'onBehalfOf',
+                expandUnderCondition: myself
+              },
+              'ui:required': (formData) => !!formData && (formData.onBehalfOf === myself)
```
Comment (erikphansen) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Can this be moved into a helper similar to `isNotAnonymous`?

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 6]
```diff
@@ -27,18 +65,183 @@ const formConfig = {
...
+            },
+            serviceBranch: {
+              'ui:title': 'Branch of service',
+              'ui:options': {
+                hideIf: (formData) => {
```
Comment (erikphansen) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Can this be moved into a helper similar to `isNotAnonymous`?

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 7]
```diff
@@ -27,18 +65,183 @@ const formConfig = {
...
+              }
+            },
+            serviceDateRange: _.merge(dateRangeUI(
+              'Service start date',
+              'Service end date',
+              'End of service must be after start of service'
+            ),
+            {
+              'ui:options': {
+                hideIf: (formData) => {
```
Comment (erikphansen) [inline src/applications/edu-benefits/complaint-tool/config/form.js]: Can this be moved into a helper similar to `isNotAnonymous`?

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 8]
```diff
@@ -27,18 +65,183 @@ const formConfig = {
...
+            fullName: _.merge(fullNameUI, {
+              prefix: {
+                'ui:title': 'Prefix',
+                'ui:options': {
+                  widgetClassNames: 'form-select-medium'
+                }
+              },
+              first: {
+                'ui:title': 'Your first name',
+                'ui:required': (formData) => !!formData && (formData.onBehalfOf !== anonymous)
```
[the remark this answers is not in the document]
Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Good idea @erikphansen, updated.

File: src/applications/edu-benefits/complaint-tool/config/form.js [hunk 9]
```diff
@@ -27,18 +65,183 @@ const formConfig = {
...
+              },
+              last: {
+                'ui:title': 'Your last name',
+                'ui:required': (formData) => !!formData && (formData.onBehalfOf !== anonymous)
```
[the remark this answers is not in the document]
Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Good idea @erikphansen, updated.

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 5 above)
Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Good idea @erikphansen, updated.

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 6 above)
Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Good idea @erikphansen, updated.

File: src/applications/edu-benefits/complaint-tool/config/form.js (hunk 7 above)
Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/config/form.js]: Good idea @erikphansen, updated.

**Review by erikphansen — APPROVED** [review]

**Comment by peggygannon** [comment]
@rrmatthews @cehsu The breadcrumb in the screenshots above still says "Complaint"  

Can we update it to say: GI Bill School Feedback Tool.

**Comment by rrmatthews** [comment]
@peggygannon the three updated screenshots should show the proper breadcrumb now! sorry!

File: package.json [hunk 10]
```diff
@@ -233,7 +233,7 @@
     "text-loader": "^0.0.1",
     "url-search-params": "^0.10.0",
     "us-forms-system": "https://github.com/usds/us-forms-system.git#84887dc7cb0ccee2ae2c8156b5664697c2850b1d",
-    "vets-json-schema": "https://github.com/department-of-veterans-affairs/vets-json-schema.git#4e5d6a1082e8268b79cb3c433ba379ce97343bc0",
+    "vets-json-schema": "https://github.com/department-of-veterans-affairs/vets-json-schema.git#223ad037ab294b946079e351a5356c9d678b12a0",
```
Comment (cehsu) [inline package.json]: The latest version is now: 444e3aba11ac1ce27e869d24bbee20160bfa3ea6, let's make sure when we merge this that the diff indicates that we are updating the version from `4e5d6a1082e8268b79cb3c433ba379ce97343bc0` and not switching it back from `444e3aba11ac1ce27e869d24bbee20160bfa3ea6` if it has already been updated by another PR.

File: src/applications/personalization/dashboard/helpers.jsx [hunk 11]
```diff
@@ -5,7 +5,7 @@ export const formBenefits = {
   '21P-527EZ': 'Veterans pension benefits',
   '21P-530': 'burial benefits',
   '1010ez': 'health care',
-  '22-0993': 'education benefits',
+  '22-0993': 'opt out',
```
Comment (cehsu) [inline src/applications/personalization/dashboard/helpers.jsx]: It looks like this was switched back by a bad merge, as it was already updated in https://github.com/department-of-veterans-affairs/vets-website/pull/8055/files

File: src/applications/personalization/dashboard/helpers.jsx [hunk 12]
```diff
@@ -14,6 +14,7 @@ export const formBenefits = {
   '22-5495': 'education benefits',
   '40-10007': 'pre-need determination of eligibility in a VA national cemetery',
   VIC: 'Veteran ID Card',
+  'complaint-tool': 'feedback',
```
Comment (cehsu) [inline src/applications/personalization/dashboard/helpers.jsx]: We are replacing complain with feedback as it is visible to the user @lihanli @jholton @rianfowler @erikphansen should we plan to replace all instances of "complaint-tool" with "feedback-tool"?

**Comment by peggygannon** [comment]
Thanks, @rrmatthews and @cehsu The breadcrumb looks good. However, i noticed something else. 

The position of "Your middle name" moved and it is now above "Your first name" and "Your last name" ... This looks odd and I think we need to move it back to the position it was before.

**Comment by cehsu** [comment]
Thanks @peggygannon, kindly refer to the updated screenshot below.
localhost_3001_education_complaint-tool_form_applicant-information 1

**Review by peggygannon — APPROVED** [review]
@cehsu Breadcrumb still needs to say "Feedback"

**Comment by cehsu** [comment]
Thanks again @peggygannon. The breadcrumbs changes were merged into master as part of another PR, and this PR has now been merged with master to include those changes, screenshot updated again below:
localhost_3001_education_complaint-tool_form_contact-information 7

**Pull request merged**